src/components/Pages/SetupPage/SetupPage.tsx

```diff
@@ -1,42 +1,118 @@
-import { useCallback, useContext, useEffect, useState } from 'react'
+import { useCallback, useContext, useEffect, useRef, useState } from 'react'
+import classNames from 'classnames'
+import { AuthIdentity } from '@dcl/crypto'
+import { localStorageGetIdentity } from '@dcl/single-sign-on-client'
 import { Button } from 'decentraland-ui/dist/components/Button/Button'
 import { Checkbox } from 'decentraland-ui/dist/components/Checkbox/Checkbox'
 import { Field } from 'decentraland-ui/dist/components/Field/Field'
+import { WearablePreview } from 'decentraland-ui/dist/components/WearablePreview/WearablePreview'
+import { connection } from 'decentraland-connect'
 import { InputOnChangeData } from 'decentraland-ui'
+import backImg from '../../../assets/images/back.svg'
+import diceImg from '../../../assets/images/dice.svg'
 import logoImg from '../../../assets/images/logo.svg'
+import platformImg from '../../../assets/images/Platform.webp'
+import { useAfterLoginRedirection } from '../../../hooks/redirection'
+import { fetchProfile } from '../../../modules/profile'
 import { FeatureFlagsContext, FeatureFlagsKeys } from '../../FeatureFlagsProvider'
+import { deployProfileFromDefault, subscribeToNewsletter } from './utils'
 import styles from './SetupPage.module.css'
 
-export const SetupPage = () => {
-  const { initialized, flags } = useContext(FeatureFlagsContext)
+enum View {
+  RANDOMIZE,
+  FORM
+}
+
+function getRandomDefaultProfile() {
+  return 'default' + (Math.floor(Math.random() * (160 - 1 + 1)) + 1)
+}
 
+export const SetupPage = () => {
+  const [initialized, setInitialized] = useState(false)
+  const [view, setView] = useState(View.RANDOMIZE)
+  const [profile, setProfile] = useState(getRandomDefaultProfile())
   const [name, setName] = useState('')
   const [email, setEmail] = useState('')
   const [agree, setAgree] = useState(false)
-
   const [nameError, setNameError] = useState('')
   const [emailError, setEmailError] = useState('')
   const [agreeError, setAgreeError] = useState('')
-
   const [showErrors, setShowErrors] = useState(false)
+  const [deploying, setDeploying] = useState(false)
+
+  const accountRef = useRef<string>()
+  const identityRef = useRef<AuthIdentity>()
+
+  const { initialized: initializedFlags, flags } = useContext(FeatureFlagsContext)
 
-  const onNameChange = useCallback((_e: any, data: InputOnChangeData) => setName(data.value), [])
-  const onEmailChange = useCallback((_e: any, data: InputOnChangeData) => setEmail(data.value), [])
+  const redirectTo = useAfterLoginRedirection()
+
+  const onRandomize = useCallback(() => setProfile(getRandomDefaultProfile()), [])
+  const onContinue = useCallback(() => setView(View.FORM), [])
+  const onBack = useCallback(() => setView(View.RANDOMIZE), [])
+  const onNameChange = useCallback((_e: unknown, data: InputOnChangeData) => setName(data.value), [])
+  const onEmailChange = useCallback((_e: unknown, data: InputOnChangeData) => setEmail(data.value), [])
```

Comment:
I know this might be a bit annoying, but in our standards, the components receive their props as `onAction`, but the functions built to be passed to the components are written as `handleAction`:
```suggestion
  const handleRandomize = useCallback(() => setProfile(getRandomDefaultProfile()), [])
  const handleContinue = useCallback(() => setView(View.FORM), [])
  const handleBack = useCallback(() => setView(View.RANDOMIZE), [])
  const handleNameChange = useCallback((_e: unknown, data: InputOnChangeData) => setName(data.value), [])
  const handleEmailChange = useCallback((_e: unknown, data: InputOnChangeData) => setEmail(data.value), [])
```

Reply:
Done